argocd: 2.2.5 -> 2.3.1

bryanasdev000 opened the pull request:
###### Description of changes

Drop packr (https://github.com/argoproj/argo-cd/commit/3f32a750a014ba818301b3e99e94be619eb18b1f), helm and kustomize as it does not seem necessary to use cli only

maintain commit and add todo for updatescript

add cmd to subpackage per  [https://github.com/argoproj/argo-cd/blob/master/Makefile\#L222](https://github.com/argoproj/argo-cd/blob/master/Makefile/#L222)

since it builds only cmd, doCheck = false is not needed anymore, also the fix for it is only in master at the current time

note, we only build the cli from now on as i think the other componentes are a bit irelevant to nixos

drop makewrapper

reorganize phases based on [https://nixos.org/manual/nixpkgs/stable/\#ssec-controlling-phases](https://nixos.org/manual/nixpkgs/stable//#ssec-controlling-phases)

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [X] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

superherointj commented on pkgs/applications/networking/cluster/argocd/default.nix:
```diff
@@ -2,18 +2,18 @@
   pname = "argocd";
-  version = "2.2.5";
-  commit = "8f981ccfcf942a9eb00bc466649f8499ba0455f5";
-  tag = "v${version}";
```
I'd prefer if you kept `tag` or used `src.rev`. It's used 4x, it is cleaner.

superherointj replied:
Removing `tag` doesn't improve the code IMO.

thiagokokada commented on pkgs/applications/networking/cluster/argocd/default.nix:
```diff
@@ -28,13 +28,15 @@ buildGoModule rec {
       "-X ${package_url}.buildDate=unknown"
-      "-X ${package_url}.gitCommit=${commit}"
-      "-X ${package_url}.gitTag=${tag}"
+      "-X ${package_url}.gitCommit=unknown"
```
Is this recommend upstream? Otherwise I would prefer that you kept the old way.

bryanasdev000 replied:
Nope, you either remove it/do not set value or set the correct value, I manually set this value to prevent any weird issue.

bryanasdev000 replied:
https://github.com/argoproj/argo-cd/blob/master/Makefile#L133

bryanasdev000 replied:
https://github.com/argoproj/argo-cd/blob/b37eee1054e42c873699460dd5e2447c2f9fe5a6/common/version.go#L10

thiagokokada replied:
Then I think it is better that you keep the commit to be specified like before (and add maybe a comment remembering people that they need to update the commit if they update the version).

bryanasdev000 replied:
Why? I mean is one more thing to keep control and ryantm's bot does not update by default, also I think these strategies of tracking is a bit irrelevant in Nix environment (but ok outside it).

bryanasdev000 replied:
I take similar approach in my other packages, I know that is a bit "ugly" not having it in version output but well, I do not know if that is worth the "hassle".

thiagokokada replied:
Setting versioning attributes to random values is known to cause some strange issues sometimes, because you never know if people are using this do to some validation. A good example: https://github.com/NixOS/nixpkgs/pull/161801.

> I mean is one more thing to keep control and ryantm's bot does not update by default

You can always write a update script and solve those two issues.

bryanasdev000 replied:
Well, then I will do this time manually and work on that.

superherointj commented on pkgs/applications/networking/cluster/argocd/default.nix:
```diff
@@ -2,18 +2,18 @@
-  version = "2.2.5";
-  commit = "8f981ccfcf942a9eb00bc466649f8499ba0455f5";
-  tag = "v${version}";
+  version = "2.3.0";
```
```suggestion
  version = "2.3.1";
```

superherointj commented on pkgs/applications/networking/cluster/argocd/default.nix:
```diff
@@ -28,13 +28,15 @@ buildGoModule rec {
   # Log: https://gist.github.com/superherointj/79cbdc869dfd44d28a10dc6746ecb3f9
+  # TODO Remove it in next release
+  # https://github.com/argoproj/argo-cd/commit/df2d6497513630f873dc17452f5ad05aac4188d0
   doCheck = false;
```
```suggestion
  doCheck = true;
```

superherointj replied:
Actually. You can just remove this.

superherointj commented on pkgs/applications/networking/cluster/argocd/default.nix:
```diff
@@ -44,8 +46,8 @@ buildGoModule rec {
   installCheckPhase = ''
-    $out/bin/argocd version --client | grep ${tag} > /dev/null
-    $out/bin/argocd-util version --client | grep ${tag} > /dev/null
+    $out/bin/argocd version --client | grep v${version} > /dev/null
+    $out/bin/argocd-util version --client | grep v${version} > /dev/null
```
```suggestion
    $out/bin/argocd version --client | grep ${tag} > /dev/null
    $out/bin/argocd-util version --client | grep ${tag} > /dev/null
```

superherointj commented on pkgs/applications/networking/cluster/argocd/default.nix:
```diff
@@ -2,18 +2,18 @@
 buildGoModule rec {
   pname = "argocd";
-  version = "2.2.5";
-  commit = "8f981ccfcf942a9eb00bc466649f8499ba0455f5";
```
```
commit = "b65c1699fa2a2daa031483a3890e6911eac69068";
```

superherointj commented on pkgs/applications/networking/cluster/argocd/default.nix:
```diff
@@ -2,18 +2,18 @@
     repo = "argo-cd";
-    rev = tag;
-    sha256 = "sha256-wSvDoRHV4BObRL8lEpHt9oGXNB06LXdIYasRYqmM5QA=";
+    rev = "v${version}";
```
```suggestion
    rev = tag;
```

superherointj commented on pkgs/applications/networking/cluster/argocd/default.nix:
```diff
@@ -2,18 +2,18 @@
   };
 
-  vendorSha256 = "sha256-BVhts+gOM6nhcR1lkFzy7OJnainLXw5YdeseBBRF2xE=";
+  vendorSha256 = "sha256-uA9sOMuVHKRRhSGoLyoKcUYU6NxtprVUITvVC+tot1g=";
```
```suggestion
  vendorSha256 = "sha256-YijhJz7m5wy8kR9V6IHSNYjiWh7H2ph6il9nMsrePOE=";
```

superherointj commented on pkgs/applications/networking/cluster/argocd/default.nix:
```diff
@@ -2,18 +2,18 @@
-  vendorSha256 = "sha256-BVhts+gOM6nhcR1lkFzy7OJnainLXw5YdeseBBRF2xE=";
+  vendorSha256 = "sha256-uA9sOMuVHKRRhSGoLyoKcUYU6NxtprVUITvVC+tot1g=";
+
+  subPackages = [ "cmd" ];
```
As you have restricted the packages availability, you need to remove unused packages from `postInstall` as well.

bryanasdev000 replied:
Even with cmd, there is a few packages there:

```
argocd
argocd-application-controller
argocd-dex
argocd-repo-server
argocd-server
argocd-util
```

superherointj commented on pkgs/applications/networking/cluster/argocd/default.nix:
```diff
@@ -28,13 +28,15 @@ buildGoModule rec {
 
   # Test is disabled because ksonnet is missing from nixpkgs.
   # Log: https://gist.github.com/superherointj/79cbdc869dfd44d28a10dc6746ecb3f9
+  # TODO Remove it in next release
+  # https://github.com/argoproj/argo-cd/commit/df2d6497513630f873dc17452f5ad05aac4188d0
   doCheck = false;
```
```suggestion
```
As tests are working now, remove the comments and `doCheck` notation. By default, it will work.

bryanasdev000 replied:
Not needed with cmd, also the drop of ksonnet did not have a release yet.

bryanasdev000 committed:
argocd: 2.2.5 -> 2.3.1

Drop packr (https://github.com/argoproj/argo-cd/commit/3f32a750a014ba818301b3e99e94be619eb18b1f), helm and kustomize as it does not seem necessary to use cli only

maintain commit and add todo for updatescript

add cmd to subpackage per  https://github.com/argoproj/argo-cd/blob/master/Makefile\#L222

since it builds only cmd, doCheck = false is not needed anymore, also the fix for it is only in master at the current time

note, we only build the cli from now on as i think the other componentes are a bit irelevant to nixos

drop makewrapper

reorganize phases based on https://nixos.org/manual/nixpkgs/stable/\#ssec-controlling-phases

superherointj commented on pkgs/applications/networking/cluster/argocd/default.nix:
```diff
@@ -1,26 +1,25 @@
-{ lib, buildGoModule, fetchFromGitHub, packr, makeWrapper, installShellFiles, helm, kustomize }:
+{ lib, buildGoModule, fetchFromGitHub, makeWrapper, installShellFiles }:
```
```suggestion
{ lib, buildGoModule, fetchFromGitHub, installShellFiles }:
```

superherointj commented on pkgs/applications/networking/cluster/argocd/default.nix:
```diff
@@ -1,26 +1,25 @@
-  version = "2.2.5";
-  commit = "8f981ccfcf942a9eb00bc466649f8499ba0455f5";
+  version = "2.3.1";
+  # Update this commit to match the above tag
+  # TODO updadeScript for it
+  commit = "b65c1699fa2a2daa031483a3890e6911eac69068";
   tag = "v${version}";
```
```suggestion
  tag = "v${version}";
  # Update commit to match the tag above
  # TODO make updadeScript
  commit = "b65c1699fa2a2daa031483a3890e6911eac69068";
```

superherointj commented on pkgs/applications/networking/cluster/argocd/default.nix:
```diff
@@ -33,20 +32,7 @@ buildGoModule rec {
-    $out/bin/argocd version --client | grep ${tag} > /dev/null
-    $out/bin/argocd-util version --client | grep ${tag} > /dev/null
-  '';
+  nativeBuildInputs = [ makeWrapper installShellFiles ];
```
```suggestion
  nativeBuildInputs = [ installShellFiles ];
```

superherointj commented on pkgs/applications/networking/cluster/argocd/default.nix:
```diff
@@ -55,6 +41,12 @@ buildGoModule rec {
+  doInstallCheck = true;
+  installCheckPhase = ''
+    $out/bin/argocd version --client | grep ${tag} > /dev/null
+    $out/bin/argocd-util version --client | grep ${tag} > /dev/null
```
```suggestion
```

superherointj commented on pkgs/applications/networking/cluster/argocd/default.nix:
```diff
@@ -55,6 +41,12 @@ buildGoModule rec {
+  '';
+
   postInstall = ''
     for appname in argocd-util argocd-server argocd-repo-server argocd-application-controller argocd-dex ; do
```
```suggestion

```
Remove the 'for appname'.

superherointj replied:
```suggestion
```

Because package is restricted to 'cmd'.  Remove:
```
for appname in argocd-util argocd-server argocd-repo-server argocd-application-controller argocd-dex ; do
   makeWrapper $out/bin/argocd $out/bin/$appname --set ARGOCD_BINARY_NAME $appname
done
```

kalbasit commented:
> note, we only build the cli from now on as i think the other componentes are a bit irelevant to nixos

What if someone is running the server from NixOS? This will be a breaking change for them. Should we split the package into `-cli`, `-server` etc..?

superherointj commented:
> > note, we only build the cli from now on as i think the other componentes are a bit irelevant to nixos
> 
> What if someone is running the server from NixOS? This will be a breaking change for them. Should we split the package into `-cli`, `-server` etc..?

At first, I did consider that. But the service is meant to be running inside the kubernetes cluster. I don't know of a case of running it outside the Kubernetes cluster. Still if it ever makes sense the person that finds the need can propose a new PR and validate everything as expected.

thiagokokada reviewed (approved):
LGTM.

Is there any other issue with the current package? If not, I am going to merge it.

superherointj commented:
> LGTM.
> 
> Is there any other issue with the current package? If not, I am going to merge it.

Go ahead. It is fine.

bryanasdev000 commented:
Also I am not using ArgoCD daily anymore because of autopilot, but I make sure to test at least the https://argo-cd.readthedocs.io/en/stable/getting_started/ after a update.
